### PR TITLE
[PDE-3678] Block promote, upload, and push for missing required app info

### PR DESCRIPTION
**Pull request by kola-er**

### CODE REVIEW

File: packages/cli/src/utils/checkMissingAppInfo.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = async function (app: Object) {
```
Comment (gregilo) [inline packages/cli/src/utils/checkMissingAppInfo.js:1]: Does this function have to be `async`? I would expect every line in this function to run synchronously.

File: packages/cli/src/oclif/commands/promote.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ const { buildFlags } = require('../buildFlags');
 const { callAPI } = require('../../utils/api');
 const { flattenCheckResult } = require('../../utils/display');
 const { getVersionChangelog } = require('../../utils/changelog');
+const checkMissingAppInfo = require('../../utils/checkMissingAppInfo');
```
Comment (gregilo) [inline packages/cli/src/oclif/commands/promote.js:10]: Good call! This is much simpler than a custom hook and more readable.

File: packages/cli/src/utils/checkMissingAppInfo.js (hunk 1 above)
Comment (gregilo) [inline packages/cli/src/utils/checkMissingAppInfo.js:1]: The other files in the `utils` directory use `kebab-case` for their file naming. Can you please update this file to match?

Comment (gregilo) [inline packages/cli/src/utils/checkMissingAppInfo.js:1]: I'm seeing an error on command line: 

We're not using TypeScript for this project and I don't think node has built-in type annotations for function parameters. Can you please remove the `: Object` in `(app: Object)`?

File: packages/cli/src/utils/checkMissingAppInfo.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+  const requiredFields = ['title', 'description', 'app_category', 'intention', 'role'];
+  const missingRequiredFields = requiredFields.filter(field => app[field] == null);
+  throw new Error(
+    `Your integration is missing required info (${missingRequiredFields.join(', ')}). Please, run "zapier register" to add it.`
+  );
```
Comment (gregilo) [inline packages/cli/src/utils/checkMissingAppInfo.js:6]: This function always throws an `Error` (even if the app has all required fields). Can you please update this to check if any required fields are missing?

Here's a suggestion: 
```suggestion
  if (missingRequiredFields.length) {
    throw new Error(
      `Your integration is missing required info (${missingRequiredFields.join(', ')}). Please, run "zapier register" to add it.`
    );
  }
```

**Review by gregilo — CHANGES_REQUESTED** [review]

File: packages/cli/src/utils/checkMissingAppInfo.js (hunk 3 above)
Comment (kola-er) [reply packages/cli/src/utils/checkMissingAppInfo.js:6]: Of course! 😄

**Review by gregilo — APPROVED** [review]

**Comment by gregilo** [comment]
@kola-er Looks good - nice work! Let's hold off on merging this until our `zapier register` updates are merged.

File: packages/cli/src/utils/build.js [hunk 4]
```diff
@@ -497,6 +499,7 @@ const buildAndOrUpload = async (
   let app;
   if (upload) {
     app = await getWritableApp();
+    checkMissingAppInfo(app);
```
Comment (eliangcs) [inline packages/cli/src/utils/build.js:502]: We probably don't want to do this in `build`. `zapier build` only generates the zip file and doesn't save the app. We should allow the developer to build the zip even if there's missing app info.

**Pull request merged**